lib/data_layer.ex

```diff
@@ -2356,14 +2358,22 @@ defmodule AshPostgres.DataLayer do
 
       case user_constraint do
         %{field: field, error_message: error_message, type: type, constraint: constraint} ->
-          Ash.Error.Changes.InvalidAttribute.exception(
-            field: field,
-            message: error_message,
-            private_vars: [
-              constraint: constraint,
-              constraint_type: type
-            ]
-          )
+          identity = Enum.find(identities, fn identity ->
+            String.contains?(constraint, to_string(identity.name))
```

Comment:
It's probably not the right approach to match violated constraint with an identity but it was the quickest way to test this feature in my local env.

Reply:
We have some logic elsewhere to get the name for a given identity, so we should reverse engineer it here. I believe we do this in the migration generator and elsewhere when we actually add in the constraint. So we can search the identities for a match on that name instead of just containing the identity name 😄